src/world/action/land.c

```diff
@@ -1,9 +1,206 @@
 #include "common.h"
+#include "functions.h"
 
-INCLUDE_ASM(void, "world/action/land", func_802B6000_E24920, void);
+void func_802B62CC_E24BEC(void);
+void func_802B644C_E24D6C(void);
 
-INCLUDE_ASM(void, "world/action/land", func_802B61C0_E24AE0, void);
+void func_802B6000_E24920(void) {
+    f32 inputMoveMagnitude;
+    f32 inputMoveAngle;
+    s32 jumpInputCheck;
+    s32 phi_a0;
+    CollisionStatus* currentCollisionStatus = &gCollisionStatus;
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    Camera* currentCameras = &gCameras[0];
+    
+    if ((playerStatus->animFlags & 0x1000) != 0) {
+        func_802B62CC_E24BEC();
+        return;
+    }
 
-INCLUDE_ASM(void, "world/action/land", func_802B62CC_E24BEC, void);
+    if (playerStatus->flags < 0) {
+        playerStatus->flags = playerStatus->flags & 0x7F77FFF1;
+        playerStatus->fallState = 0;
+        playerStatus->decorationList = 0;
+        playerStatus->unk_C2 = 0;
+        playerStatus->decorationPos[0] = playerStatus->position.x;
+        playerStatus->decorationPos[1] = playerStatus->position.z;
 
-INCLUDE_ASM(void, "world/action/land", func_802B644C_E24D6C, void);
+        if ((playerStatus->animFlags & 0x4000) != 0) {
+            phi_a0 = 0x9 << 16;
+            phi_a0 |= 0x2;
+
+        } else {
+            phi_a0 = 0x6000B;
+            if ((playerStatus->animFlags & 1) == 0) {
+                phi_a0 = 0x1 << 16;
+                phi_a0 |= 0x9;
```

Comment:
Can this line be merged with the one above it? Preferably it'd be something like 1 << 16 | 1<< 3 | 1 << 1; and maybe those correspond to one of our flag defines

Reply:
Based on the function below (suggest_player_anim_clearUnkFlag), I assume those are animFlags but I'm not certain.  Should I just go with that for now?

Reply:
I'd leave them raw for now, and wait for someone else to chime in on that, I'm not sure as well

Reply:
Ok, also it seems that line cannot be merged with the above one.

Reply:
Usually I find if the decompiler had them split like that, they need to be that way. Surprisingly GCC doesn't optimize this.

Reply:
Should I leave a comment stating that this area should be cleaned up in the future?

Reply:
It's cool to just leave it as-is. But could you make the 0x1 << 16 just a literal value instead of the shifted version? We write our flags with the full hex literal

Reply:
Yep :+1: